src/Compilers/VisualBasic/Portable/Binding/AttributeSemanticModel.vb

```diff
@@ -26,7 +26,26 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         ''' </summary>
         Friend Shared Function Create(containingSemanticModel As SyntaxTreeSemanticModel, binder As AttributeBinder, Optional ignoreAccessibility As Boolean = False) As AttributeSemanticModel
             Debug.Assert(containingSemanticModel IsNot Nothing)
-            Return New AttributeSemanticModel(binder.Root, binder, containingSemanticModel, ignoreAccessibility:=ignoreAccessibility)
+            Dim owner As Symbol = GetAttributeTarget(containingSemanticModel, binder)
+            Dim wrappedBinder As Binder = binder
+            If owner IsNot Nothing Then
+                wrappedBinder = New LocationSpecificBinder(BindingLocation.Attribute, owner, binder)
+            End If
+
+            Return New AttributeSemanticModel(binder.Root, wrappedBinder, containingSemanticModel, ignoreAccessibility:=ignoreAccessibility)
+        End Function
+
+        Private Shared Function GetAttributeTarget(model As SyntaxTreeSemanticModel, binder As AttributeBinder) As Symbol
+            If binder.AttributeTarget IsNot Nothing Then
```

Comment:
When is this false?

Reply:
It's probably always false actually.

See

https://github.com/dotnet/roslyn/blob/b55ec310613097be69bc8beb4b34f810771d0fa2/src/Compilers/VisualBasic/Portable/Binding/BinderBuilder.vb#L298

and

https://github.com/dotnet/roslyn/blob/b55ec310613097be69bc8beb4b34f810771d0fa2/src/Compilers/VisualBasic/Portable/Binding/BinderBuilder.vb#L305

The second one is the semantic model path, which isn't aware of a symbol, only a syntax node.

I added the check defensively in case it's possible to get here with a non-null symbol and a null syntax node.